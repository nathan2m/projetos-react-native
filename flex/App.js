import React from 'react';
import {StatusBar, StyleSheet, Text, View} from 'react-native';

export default function App() {
  return (
    <View style={styles.container}>
      <StatusBar />
      {/* <View style={styles.box} />
        <View style={styles.box} />
        <View style={styles.box} />
        <View style={styles.box} />
        <View style={styles.box} /> */}
      <View style={styles.header}>
        <Text>Voltar</Text>
        <Text>Titulo</Text>
        <Text>Perfil</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#CCC',
    /* flexDirection: 'row',
    flexWrap: 'wrap',
    alignContent: 'space-between', */
  },

  header: {
    height: 60,
    backgroundColor: '#fff',
    //paddingTop: 20,
    paddingHorizontal: 20,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },

  /* box: {
    flex: 1,
    minWidth: 100,
    height: 100,
    backgroundColor: '#666',
    borderColor: '#999',
    borderWidth: 5,
    margin: 10,
  }, */
});
