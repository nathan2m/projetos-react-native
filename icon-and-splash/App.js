//import { StatusBar } from 'expo-status-bar';
import React, { Component } from "react";
import { 
    StatusBar, 
    StyleSheet, 
    Text, 
    View 
} from "react-native";

export default class App extends Component {
    render() {
        return (
            <View style={styles.container}>
                <StatusBar
                    barStyle="light-content"
                    backgroundColor={styles.container.backgroundColor}
                />
                <Text style={styles.welcome}>Welcome to Academy!</Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "#7159C1"
    },
    welcome: {
        fontSize: 20,
        textAlign: "center",
        margin: 10,
        color: "#FFF",
        fontWeight: "bold"
    }
});