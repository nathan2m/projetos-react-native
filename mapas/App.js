import React, {Component} from 'react';
import {ScrollView, View, Text, StyleSheet, Dimensions} from 'react-native';

import MapView from 'react-native-maps';

export default class App extends Component {
  state = {
    places: [
      {
        id: 1,
        title: 'Casa do café',
        description: 'Café quentinho...',
        latitude: -27.210671,
        longitude: -49.63627,
      },
      {
        id: 2,
        title: 'RocketSeat',
        description: 'Programação, empreendedorismo e mindset',
        latitude: -27.200671,
        longitude: -49.63627,
      },
      {
        id: 3,
        title: 'Casa do José',
        description: 'José, tá em casa?',
        latitude: -27.200671,
        longitude: -49.62627,
      },
    ],
  };

  _mapReady = () => {
    this.state.places[0].mark.showCallout();
  };

  render() {
    const {latitude, longitude} = this.state.places[0];

    return (
      <View style={styles.container}>
        <MapView
          ref={map => (this.mapView = map)}
          initialRegion={{
            latitude,
            longitude,
            latitudeDelta: 0.0142,
            longitudeDelta: 0.0231,
          }}
          style={styles.mapView}
          rotateEnabled={false}
          scrollEnabled={false}
          zoomEnabled={false}
          showsPointsOfInterest={false}
          showBuildings={false}
          onMapReady={this._mapReady}>
          {this.state.places.map(place => (
            <MapView.Marker
              ref={mark => (place.mark = mark)}
              title={place.title}
              description={place.description}
              key={place.id}
              coordinate={{
                latitude: place.latitude,
                longitude: place.longitude,
              }}
            />
          ))}
        </MapView>
        <ScrollView
          style={styles.placesContainer}
          horizontal
          pagingEnabled
          showsHorizontalScrollIndicator={false}
          onMomentumScrollEnd={e => {
            const place =
              e.nativeEvent.contentOffset.x > 0
                ? e.nativeEvent.contentOffset.x / Dimensions.get('window').width
                : 0;

            const {latitude, longitude, mark} = this.state.places[place];

            this.mapView.animateToCoordinate(
              {
                latitude,
                longitude,
              },
              500,
            );

            setTimeout(() => {
              mark.showCallout();
            }, 500);
          }}>
          {this.state.places.map(place => (
            <View key={place.id} style={styles.place}>
              <Text style={styles.title}>{place.title}</Text>
              <Text style={styles.description}>{place.description}</Text>
            </View>
          ))}
        </ScrollView>
      </View>
    );
  }
}

const {height, width} = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
  },

  mapView: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
  },

  placesContainer: {
    width: '100%',
    maxHeight: 200,
  },

  place: {
    width: width - 40,
    maxHeight: 200,
    backgroundColor: '#FFF',
    marginHorizontal: 20,
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    padding: 20,
  },

  title: {
    fontWeight: 'bold',
    fontSize: 18,
    backgroundColor: 'transparent',
  },

  description: {
    color: '#999',
    fontSize: 12,
    marginTop: 5,
  },
});
