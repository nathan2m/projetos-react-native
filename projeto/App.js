import React, { Component } from 'react';

import { 
  StyleSheet, 
  Text, 
  View, 
  ScrollView, 
} from 'react-native';

export default class App extends Component {
  render() {
    return (
      <ScrollView style={styles.container}>
        <View style={styles.postContainer}>
          <Text style={styles.postTitle}>Estudando React Native</Text>
          <Text style={styles.postDescription}>Lorem ipsum sit dolot amet....</Text>
        </View>
        
        <View style={styles.postContainer}>
          <Text style={styles.postTitle}>Estudando React Native</Text>
          <Text style={styles.postDescription}>Lorem ipsum sit dolot amet....</Text>
        </View>
        
        <View style={styles.postContainer}>
          <Text style={styles.postTitle}>Estudando React Native</Text>
          <Text style={styles.postDescription}>Lorem ipsum sit dolot amet....</Text>
        </View>
        
        <View style={styles.postContainer}>
          <Text style={styles.postTitle}>Estudando React Native</Text>
          <Text style={styles.postDescription}>Lorem ipsum sit dolot amet....</Text>
        </View>
        
        <View style={styles.postContainer}>
          <Text style={styles.postTitle}>Estudando React Native</Text>
          <Text style={styles.postDescription}>Lorem ipsum sit dolot amet....</Text>
        </View>
        
        <View style={styles.postContainer}>
          <Text style={styles.postTitle}>Estudando React Native</Text>
          <Text style={styles.postDescription}>Lorem ipsum sit dolot amet....</Text>
        </View>
        
        <View style={styles.postContainer}>
          <Text style={styles.postTitle}>Estudando React Native</Text>
          <Text style={styles.postDescription}>Lorem ipsum sit dolot amet....</Text>
        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#333',
  },
  postContainer: {
    marginHorizontal: 20,
    marginVertical: 10,
    padding: 20,
    backgroundColor: '#fff',
    borderRadius: 3,
  },
  postTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 5,
  },
  postDescription: {
    color: '#666',
  },
});