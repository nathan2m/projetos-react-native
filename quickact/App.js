import React, {Component} from 'react';
import {
  Platform,
  DeviceEventEmitter,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import QuickActions from 'react-native-quick-actions';

QuickActions.setShortcutItems([
  {
    type: 'contacts', // Required
    title: 'Listar contatos', // Optional, if empty, `type` will be used instead
    subtitle: 'Ver amigos',
    icon: Platform.OS === 'ios' ? 'Rocket' : 'rocket_icon', // Pass any of UIApplicationShortcutIconType<name>
    userInfo: {
      url: 'app://contacts', // provide custom data, like in-app url you want to open
    },
  },
]);

DeviceEventEmitter.addListener('quickActionShortcut', data => {
  console.log(data);
});

QuickActions.popInitialAction().then(data => {
  console.log(data);
});

export default class App extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to React Native!</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});
