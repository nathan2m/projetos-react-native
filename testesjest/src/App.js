import React, {Component} from 'react';

import {StyleSheet, Text, View} from 'react-native';

import Todos from './Todos';

class App extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>Hello Test</Text>
        <Todos />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignContent: 'center',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default App;
