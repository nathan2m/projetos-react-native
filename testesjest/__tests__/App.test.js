import 'react-native';
import React from 'react';
import App from '../src/App';
import {shallow} from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<App />);
});

/* describe('Testing App Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
}); */
