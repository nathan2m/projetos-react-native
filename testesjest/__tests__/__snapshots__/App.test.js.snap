// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component should render correctly 1`] = `ShallowWrapper {}`;
